#!/usr/bin/env python
PACKAGE = "tiago_pcl_tutorial"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("enableDownSampling",     bool_t,   0, "Enable downsampling",                   True)
gen.add("downSamplingLeafSize",   double_t, 0, "Leaf size",                             0.01, 0.005,      0.9)
gen.add("minNeighbours",          int_t,    0, "Min number of neighbourds",               30,     3,      100)
gen.add("minClusterSize",         int_t,    0, "Min number of points in cluster",         50,     1,     1000)
gen.add("maxClusterSize",         int_t,    0, "Max number of points in cluster",    1000000,   100,  1000000)
gen.add("smoothessThreshold",     double_t, 0, "Smoothness threshold in degrees",          3,     0,      180)
gen.add("curvatureThreshold",     double_t, 0, "Curvature threshold",                    1.0,   0.0,      5.0)

exit(gen.generate(PACKAGE, "tiago_pcl_tutorial", "region"))
